boards/arm/nucleo_h743zi/pinmux.c

```diff
@@ -14,6 +14,37 @@
 
 /* NUCLEO-H743ZI pin configurations */
 static const struct pin_config pinconf[] = {
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(usart3), okay) && CONFIG_SERIAL
+	{ STM32_PIN_PD8, STM32H7_PINMUX_FUNC_PD8_USART3_TX },
+	{ STM32_PIN_PD9, STM32H7_PINMUX_FUNC_PD9_USART3_RX },
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(i2c1), okay) && CONFIG_I2C
+	{ STM32_PIN_PB8, STM32H7_PINMUX_FUNC_PB8_I2C1_SCL },
+	{ STM32_PIN_PB9, STM32H7_PINMUX_FUNC_PB9_I2C1_SDA },
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(pwm12), okay) && CONFIG_PWM
+	{ STM32_PIN_PB14, STM32H7_PINMUX_FUNC_PB14_PWM12_CH1 },
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(adc1_2), okay) && CONFIG_ADC
+	{ STM32_PIN_PA3, STM32H7_PINMUX_FUNC_PA3_ADC12_INP15 },
+#endif
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(spi1), okay) && CONFIG_SPI
+#ifdef CONFIG_SPI_STM32_USE_HW_SS
+	{STM32_PIN_PD14,  STM32H7_PINMUX_FUNC_PD14_SPI1_NSS},
+#endif /* CONFIG_SPI_STM32_USE_HW_SS */
+	{STM32_PIN_PA5,  STM32H7_PINMUX_FUNC_PA5_SPI1_SCK},
+	{STM32_PIN_PA6, STM32H7_PINMUX_FUNC_PA6_SPI1_MISO},
+	{STM32_PIN_PB5, STM32H7_PINMUX_FUNC_PB5_SPI1_MOSI},
+#endif
```

Comment:
Thanks for your contribution, but you should probably rebase your branch on master since UART I2C, PWM and SPI are now defined from device tree and no longer from pinmux.c.

I suggest the following:

```suggestion

```